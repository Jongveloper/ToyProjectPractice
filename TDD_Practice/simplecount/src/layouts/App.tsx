import React, { useCallback, useState } from 'react';

const App = () => {
  const [count, setCount] = useState(0);

  const plusCount = useCallback(() => {
    setCount((count) => count + 1)
  }, [])

  const minusCount = useCallback(() => {
    setCount((count) => count - 1);
  }, [])
  return (
    <div>
      <h3 data-testid='counter'>{count}</h3>
      <div>
        <button data-testid='plus-button' onClick={plusCount}>+</button>
        <button data-testid='minus-button' onClick={minusCount}>-</button>
      </div>
    </div>
  );
}

export default App;
