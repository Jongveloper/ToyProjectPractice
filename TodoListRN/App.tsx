import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';

import AppInner from './AppInner';
import { Provider } from "react-redux";
import store from "./src/store";

function App() {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <AppInner />
      </NavigationContainer>
    </Provider>
  );
}

export default App;