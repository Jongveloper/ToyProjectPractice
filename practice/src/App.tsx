import React, { useState } from 'react';

import Form from 'components/Form';
import List from 'components/List';

export interface Task {
  id: number;
  title: string;
}

interface Tasks {
  newId: number;
  taskTitle: string;
  tasks: Task[];
}

const App = () => {
  const [state, setState] = useState<Tasks>({
    newId: 100,
    taskTitle: '',
    tasks: [],
  });

  const { newId, taskTitle, tasks } = state;

  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
    setState({
      ...state,
      taskTitle: event.target.value,
    });
  };

  const handleClickAddTask = () => {
    setState({
      ...state,
      newId: newId + 1,
      tasks: [...tasks, { id: newId, title: taskTitle }],
      taskTitle: '',
    });
  };

  const handleClickDeleteTask = (id: number) => {
    setState({
      ...state,
      tasks: tasks.filter(task => task.id !== id),
    });
  };

  return (
    <>
      <Form
        taskTitle={taskTitle}
        handleChange={handleChangeTitle}
        handleClick={handleClickAddTask}
      />
      <List tasks={tasks} handleClick={handleClickDeleteTask} />
    </>
  );
};
export default App;
