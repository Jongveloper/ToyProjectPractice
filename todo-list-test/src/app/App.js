import React, { useCallback, useState } from 'react';
import Todos from '../components/todos/todos';

function App({ presenter }) {
  const [todos, setTodos] = useState(presenter.getTodos());

  const handleAdd = useCallback((todo) => {
    presenter.add(todo, setTodos);
  }, []);

  const handleDelete = useCallback((todo) => {
    presenter.delete(todo, setTodos);
  }, []);

  console.log(todos);

  return (
    <>
      <Todos todos={todos} onAdd={handleAdd} onDelete={handleDelete} />
    </>
  );
}

export default App;
