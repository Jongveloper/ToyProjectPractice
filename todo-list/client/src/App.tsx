import React from 'react';
import './App.css';
import Header from './components/Header';
import backImg from './img/back.png'

function App() {
  return (
    <>
      <img src={backImg} alt="배경" width='100%' height='1305vh' />
      <Header />
    </>
  );
}

export default App;
