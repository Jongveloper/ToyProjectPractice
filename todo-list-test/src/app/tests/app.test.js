import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import TodoPresenter from '../todo_presenter';

describe('App', () => {
  let presenter;
  beforeEach(() => {
    presenter = new TodoPresenter([
      { id: 1, name: 'Learning React' },
      { id: 2, name: 'Learning Nodejs' },
      { id: 3, name: 'Learning Nextjs' },
    ]);
  });

  it('renders', () => {
    const component = renderer.create(<App presenter={presenter} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  const setup = () => render(<App presenter={presenter} />);

  it('todo를 추가합니다', () => {
    setup();
    const newTodo = 'Learning JavaScript';
    const input = screen.getByPlaceholderText('해야할 일을 입력하세요');
    const button = screen.getByText('Add');
    userEvent.type(input, newTodo);
    userEvent.click(button);
    const addedTodo = screen.getAllByTestId('todo');
    expect(addedTodo.innerHTML).toBe(newTodo);
  });

  it('todo를 삭제합니다.', () => {
    setup();
    const first = screen.getAllByTitle('삭제')[0];
    userEvent.click(first);
    const next = screen.getAllByTestId('todo')[0];
    expect(next.innerHTML).not.toBe('Learning React');
  });
});
