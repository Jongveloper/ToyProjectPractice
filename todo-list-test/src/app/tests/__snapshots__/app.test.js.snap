// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div>
  <form
    className="add-from"
    onSubmit={[Function]}
  >
    <input
      className="add-input"
      placeholder="해야할 일을 입력하세요"
      type="text"
    />
    <button
      className="add-button"
    >
      Add
    </button>
  </form>
  <ul>
    <li>
      <span
        data-testid="todo"
      >
        Learning React
      </span>
      <button
        onClick={[Function]}
        title="삭제"
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
    <li>
      <span
        data-testid="todo"
      >
        Learning Nodejs
      </span>
      <button
        onClick={[Function]}
        title="삭제"
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
    <li>
      <span
        data-testid="todo"
      >
        Learning Nextjs
      </span>
      <button
        onClick={[Function]}
        title="삭제"
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
  </ul>
</div>
`;
